### PR TITLE
Benji/2774 session frame condensing

### LINKED ISSUE #2774: Refactor Sesssion Modal Common Controls
Each `TMSessionXYZ.vue` component has common functionality:

- Title
- Close
- Go back

This should be factored out into a common component `TmSessionFrame.vue` to avoid repeating ourselves.

At least these to start with:

- TmSessionWelcome
- TmSessionSignUp
- TmSessionExisting
- TmSessionExplore
- TmSessionHardware
- TmSessionExtension

### PR DESCRIPTION
**Pull request by thebkr7**
Closes #2774 

**Description:**

Condenses duplicate TmSession components functionality into a single component.

- TmSessionWelcome
- TmSessionSignUp
- TmSessionExisting
- TmSessionExplore
- TmSessionHardware
- TmSessionExtension

Thank you! 🚀

---

For contributor:

- [x] Added changes entries. Run `yarn changelog` for a guided process.
- [x] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by thebkr7** [comment]
Will finish updating the tests before marking Ready for Review.

**Comment by jbibla** [comment]
this is pretty nice!

File: src/components/common/SessionFrame.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      default: true
+    },
+    nestedBack: {
+      type: Boolean,
+      default: true
+    }
+  },
+  methods: {
+    setState(value) {
+      this.$parent.$parent.$emit(`route-change`, value)
```
Comment (jbibla) [inline src/components/common/SessionFrame.vue]: hmmm this doesn't smell right 👃 

@colw what's your take?

File: src/components/common/SessionFrame.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    close() {
+      this.$parent.$parent.$emit(`close`)
+    },
+    goBack() {
+      let shortenedKey = this.$parent
+      if (this.nestedBack) {
+        shortenedKey = this.$parent.$parent
```
Comment (jbibla) [inline src/components/common/SessionFrame.vue]: this is also a red flag for me

File: src/components/common/SessionFrame.vue (hunk 1 above)
Comment (jbibla) [reply src/components/common/SessionFrame.vue]: https://vuejs.org/v2/guide/components-edge-cases.html#Accessing-the-Parent-Component-Instance

Comment (thebkr7) [reply src/components/common/SessionFrame.vue]: Fully agree. There are 2 things that I will improve on.
1. Those 3 props could become only one.
2. Close function could be passed down from SessionRouter.

**Comment by thebkr7** [comment]
Putting this PR on pause until we decide on the routing architecture. Commented the areas in SessionFrame that will need to be removed in favour of router-link. The tests for the TmSessions will need to take into account the router-link and have not been updated fully.

**Comment by thebkr7** [comment]
The plan is to remove SessionRouter completely and rather have a route in routes.js that users are sent to when the click the Sign in button. Users then can be sent back to the screen they were on before with route history. The goBack and Close buttons would also harness the route history thus simplifying the logic in SessionFrame.

**Pull request closed without merging**